### PR TITLE
Michael/replace syntax to esm

### PR DESCRIPTION
**Pull request by mikicho**
Phase 1 of https://github.com/practicajs/practica/issues/12
* Update the syntax
* Fix types

Open issue:
Some props in `interactive-cli.tsx` don't exist in the definition files, one of them is wrong, we need to find out which and fix then

### CODE REVIEW

**Comment by goldbergyoni** [comment]
@mikicho Awesome awesome PR 🛰

p.s. Any idea why do the tests fail?

**Comment by mikicho** [comment]
> @mikicho Awesome awesome PR 🛰
> 
> p.s. Any idea why do the tests fail?

@goldbergyoni see the open issue (TL;DR types) but basically, the tests pass

**Comment by mikicho** [comment]
@goldbergyoni Fixed

**Comment by goldbergyoni** [comment]
@giltayar You should be proud of @mikicho who takes us to ESM land:)

p.s. I hope he stops here and won't migrate our tests to Mocha 😧

File: src/code-templates/libraries/error-handling/error-handling.ts [hunk 1]
```diff
@@ -25,10 +25,8 @@ const decideWhetherToCrash = (error) => {
 };
 
 class AppError extends Error {
-  constructor(name, message, public HTTPStatus, public isTrusted) {
+  constructor(readonly name: string, readonly message: string, public readonly HTTPStatus, public readonly isTrusted = true) {
```
Comment (goldbergyoni) [inline src/code-templates/libraries/error-handling/error-handling.ts:28]: Yeah, good addition

File: src/code-templates/services/order-service/data-access/order-repository.ts [hunk 2]
```diff
@@ -16,23 +16,23 @@ export default class OrderRepository {
...
         },
         externalIdentifier: {
-          type: Sequelize.STRING,
+          type: DataTypes.STRING,
           unique: true,
           allowNull: true,
         },
         mode: {
-          type: Sequelize.STRING,
+          type: DataTypes.STRING,
```
Comment (goldbergyoni) [inline src/code-templates/services/order-service/data-access/order-repository.ts:29]: @mikicho How does it fell like working with ORM 😅

**Review by goldbergyoni — APPROVED** [review]
Super-awesome PR 🛰

**Comment by giltayar** [comment]
@goldbergyoni great progress! But until TypeScript doesn't compile the code to ESM, I won't call it done. 😉

BTW, why is the target compilation `es5`? It can be easily `es2020`

**Comment by mikicho** [comment]
@giltayar Soon :)
https://devblogs.microsoft.com/typescript/announcing-typescript-4-7-rc/#esm-nodejs

**Comment by goldbergyoni** [comment]
@giltayar Yeah good catch, the generated code has es2020 but the root project for some reason was left with es5

@mikicho Maybe to avoid regression, until we pull the plug and go full-ESM, include a listing rule 'no-require'?

**Comment by mikicho** [comment]
IMHO, I dont think this is necessary.

**Comment by mikicho** [comment]
@giltayar @goldbergyoni https://mobile.twitter.com/typescript/status/1529194086741069824 :)

**Comment by goldbergyoni** [comment]
Niceeee

Though, I would wait a while before making the switch

**Pull request merged**